src/DocumentScope.zig

```diff
@@ -156,6 +156,7 @@ pub const Declaration = union(enum) {
     };
 
     pub const Switch = struct {
+        identifier: Ast.TokenIndex,
```

Comment:
This unnecessarily increases the size of each DocumentScope Declaration from 8 to 12 bytes for the untagged union.

Having two different fields in `Declaration` should work for this:

```zig
/// - `switch (condition) { .case => |value| {} }`
switch_payload: Switch,
/// - `switch (condition) { inline .case => |_, tag| {} }`
/// - `switch (condition) { inline else => |_, tag| {} }`
switch_inline_tag_payload: Switch,
```